query.go

```diff
@@ -758,15 +795,22 @@ func (s *ChainService) GetCFilter(blockHash chainhash.Hash,
 		options...,
 	)
 
-	// If we've found a filter, write it to the database for next time.
 	if filter != nil {
-		err := s.FilterDB.PutFilter(&blockHash, filter, dbFilterType)
-		if err != nil {
-			return nil, err
-		}
+		// If we found a filter, put it in the cache and persistToDisk if
+		// the caller requested it.
+		s.putFilterToCache(&blockHash, dbFilterType, filter)
```

Comment:
check returned error

Reply:
what do you suggest to do on error ?

Reply:
Like I said above, I don't think filterdb and cache are critical components. By this point, we already have a filter, why fail the call ? Insertion into cache could fail for different reasons. For example, what if the filter is bigger than the cache's size (unlikely, but still...), that would mean that GetCFilter would never be able to return that filter.

Reply:
Good point. I'm fine with just logging the error, if we make sure we are not relying on being able to add to the cache :)